python/tests/res/simulator/test_simulation_context.py

```diff
@@ -37,30 +35,19 @@ def test_simulation_context(self):
             simulation_context1 = SimulationContext(ert, first_half, mask1 , 0)
             simulation_context2 = SimulationContext(ert, other_half, mask2 , 0)
 
-            ert.createRunpath( simulation_context1.get_run_context( ) )
-            ert.createRunpath( simulation_context2.get_run_context( ) )
-
             geo_id = 0
             for iens in range(size):
                 if iens % 2 == 0:
-                    simulation_context1.addSimulation(iens, geo_id)
                     self.assertFalse(simulation_context1.isRealizationFinished(iens))
                 else:
-                    simulation_context2.addSimulation(iens, geo_id)
                     self.assertFalse(simulation_context2.isRealizationFinished(iens))
 
+            wait_until(
+                func=(lambda: self.assertFalse(simulation_context1.isRunning() or simulation_context2.isRunning())),
+                interval=1
+            )
 
-            with self.assertRaises(UserWarning):
-                simulation_context1.addSimulation(size, geo_id)
-
-            with self.assertRaises(UserWarning):
-                simulation_context1.addSimulation(0, geo_id)
-
-            while simulation_context1.isRunning():
-                time.sleep(1.0)
-
-            while simulation_context2.isRunning():
-                time.sleep(1.0)
+            time.sleep(3.0)
```

Comment:
Please remove the `time.sleep` from all the tests and lower the `wait_until` interval to `0.1`. This should not appear in a test. If this is necessary for the code to work we should figure out why.

Reply:
`time.sleep()` is unfortunately required to sync the `fs` with job states. We could perphaps integrate it into the `wait_until`

Reply:
This's been fixed now by waiting for the `enkf_simulation_runner.runSimpleStep` to finish.